psi4/driver/driver.py

```diff
@@ -1948,6 +1948,45 @@ def fchk(wfn, filename):
 
     """
     fw = core.FCHKWriter(wfn)
+
+    # idea: get the method from the variable matching closely the 'current energy'
+    #below is the first try, not pretty
+    varlist = core.variables()
+    current = varlist['CURRENT ENERGY']
+    del varlist['CURRENT ENERGY']
+
+    # delete disturbing entries
+    delete_list = ['CURRENT ENERGY', 'CURRENT GRADIENT', 'CURRENT REFERENCE ENERGY']
+    for key in delete_list[:]:
+        if key in varlist:
+            del varlist[key]
+
+    # find closest match
+    for (key, val) in varlist.items():
+        if hasattr(val, '__len__'): continue
+        close = np.isclose(val, current)  # needs improvement
+        # print(close,key,val,current) # debug, remove
+        if (close):
+            # print(key,val) # debug, remove
+            method = key
+            break
+    print('method key:', method)  # debug, remove
+
+    fchk_label = ' SCF Density'
+    if (re.search("CC", method) is not None):
+        fchk_label = ' CC Density'
+    elif (re.search("MP2", method) is not None):
+        fchk_label = ' MP2 Density'
+    elif (re.search("MP3", method) is not None):
+        fchk_label = ' MP3 Density'
+    elif (re.search("MP4", method) is not None):
+        fchk_label = ' MP4 Density'
+    elif (re.search("CI", method) is not None):
+        fchk_label = ' CI Density'
+    elif (re.search("DC", method) is not None):
```

Comment:
All DCT psivars use the letters "[DCT](http://www.psicode.org/psi4manual/master/autodir_psivariables/module__dct.html#apdx-dct-psivar)", so you can do that for a more specific search.

Also, is there a reason you're using a regex here? All of your `re.search(search_string, method) is not None` can be written as `search_string in method`. Less regex, less code.

Reply:
thx. Initially I thought it would be more difficult to map `method` to the `label`. The `wfn.name()` could really have been anything. Now with the psivars also that got simpler and regex is likely not needed, true.